sklearn/linear_model/tests/test_base.py

```diff
@@ -205,6 +208,25 @@ def test_linear_regression_sparse_multiple_outcome(random_state=0):
     assert_array_almost_equal(np.vstack((y_pred, y_pred)).T, Y_pred, decimal=3)
 
 
+def test_linear_regression_pd_sparse_dataframe_warning():
+    try:
+        import pandas as pd
+        if LooseVersion(pd.__version__) < '0.24.0':
+            return
+        df = pd.DataFrame()
+        for col in range(4):
+            arr = np.random.randn(10)
+            arr[:8] = 0
+            arr = pd.SparseArray(arr)
+            df[str(col)] = pd.SparseArray(arr, fill_value=0)
```

Comment:
Is the defining `arr` twice needed?

```py
arr[:8] = 0
df[str(col)] = pd.arrays.SparseArray(arr, fill_value=0)
```

(Using `pd.arrays.SparseArray` because `pd.SparseArray` will be deprecated)

Reply:
oh, I am sorry. That's a mistake.